fxa/oauth.py

```diff
@@ -177,3 +178,55 @@ def verify_token(self, token, scope=None):
                 raise ScopeMismatchError(authorized_scope, scope)
 
         return resp
+
+
+class MemoryCache(object):
+    """Simple Memory cache that caches the object in the memory."""
+
+    def __init__(self, ttl):
+        self.ttl = ttl
+        if not self.ttl:
+            self.ttl = DEFAULT_CACHE_EXPIRACY
+        self.cache = {}
+        self.expires_at = {}
+
+    def get(self, key):
+        value = self.cache.get(key, None)
+        expires_at = self.expires_at(key, None)
+        if expires_at and expires_at > time.time():
+            return value
+        else:
+            self.delete(key)
+
+    def set(self, key, value):
+        self.cache[key] = value
+        self.expires_at[key] = time.time() + self.ttl
+
+    def delete(self, key):
+        if key in self.cache:
+            del self.cache[key]
+
+        if key in self.expires_at:
+            del self.expires_at[key]
+
+
+class CachedClient(Client):
+    """Simple cache client that uses the cache to cache the data."""
```

Comment:
I believe the description should be more accurate. Also, you're using three times "cache".